src/libraries/System.Private.CoreLib/src/System/IO/Directory.Unix.cs

```diff
@@ -20,5 +25,24 @@ private static DirectoryInfo CreateDirectoryCore(string path, UnixFileMode unixC
 
             return new DirectoryInfo(path, fullPath, isNormalized: true);
         }
+
+        private static unsafe string CreateTempSubdirectoryCore(string? prefix)
+        {
+            // mkdtemp takes a char* and overwrites the XXXXXX with six characters
+            // that'll result in a unique file name.
+            string template = $"{Path.GetTempPath()}{prefix}XXXXXX\0";
+            byte[] name = Encoding.UTF8.GetBytes(template);
```

Comment:
Can we avoid this temporary byte[] allocation? e.g. something like:
```C#
string tempPath = Path.GetTempPath();
int length = tempPath.Length + (prefix?.Length ?? 0) + 6 + 1;
Span<byte> path = length <= 256 ? stackalloc byte[256] : new byte[length];
int pos = Encoding.UTF8.GetBytes(tempPath, path);
if (prefix is not null)
{
    pos += Encoding.UTF8.GetBytes(prefix, path.Slice(pos));
}
path.Slice(pos, 6).Fill((byte)'X');
path[pos + 6] = 0;
```

Reply:
Done. I also changed Path.GetTempFileName, since this code was cloned from there.